src/orion/core/io/cmdline_parser.py

```diff
@@ -0,0 +1,229 @@
+        """Parse the `commandline` argument.
+
+        Create an OrderedDict where the keys are the names of the arguments and the values
+        are the actual values of the each argument. The arguments can be a single value or a
+        list of values. This also supports positional arguments.
+
+        Parameters
+        ----------
+        commandline : list
+            List of string representing the commmandline arguments.
+
+        Returns
+        -------
+        OrderedDict
+            Dictionary holding the values of every argument. The keys are the arguments' name.
+
+        Raises
+        ------
+        ValueError
+            If there is a duplicate argument
+
+        Notes
+        -----
+        By default, all values are `str` unless their types have been changed in the meantime.
+        Unnamed arguments's key follow the format '_X' where `X` is the index of that argument
+        in the list. For example, `val1 val2` would be parsed as
+        `{'_pos_0': 'val1', '_pos_1': 'val2}`.
+
+        Arrays are parsed starting at the first named argument until the next one. For example :
+        `--arg1 value1 value2 --arg2 value3 value4` will be parsed as :
+        `{'arg1': ['value1', 'value2'], 'arg2': ['value3', 'value4']}`.
+
+        File paths are extended to their absolute forms.
+
+        If the commandline contains optional arguments before a subcommand, the arguments
+        will be wrongly parsed : `somecommand --optional argument subcommand --another argument`
+        will be parsed as `{'optional': ['argument', 'subcommand'], 'another': 'argument'}`
+
+        When dealing with an empty list (of the form `--args`), the value is a boolean.
+        This indicates to the parsing process to skip this particular argument.
+
+        """
+        if not commandline:
+            return self.arguments
+
+        self.arguments = OrderedDict()
+        self._parse_arguments(commandline)
+
+        for key, value in self.arguments.items():
+            # Handle positional arguments
+            if key.startswith("_"):
+                self.template.append("{" + key + "}")
+
+            # Handle optional ones
+            else:
+                key_name = self._key_to_arg(key)
+
+                if key_name in self.template:
+                    continue
+
+                self.template.append(key_name)
+
+                # Ignore value as key is a boolean argument
+                if isinstance(value, bool):
+                    continue
+
+                if not isinstance(value, list):
+                    template = "{" + key + "}"
+                    self.template.append(template)
+                    continue
+
+                for pos in range(len(value)):
+                    template = "{" + key + "[" + str(pos) + "]}"
+                    self.template.append(template)
+
+        self._already_parsed = True
+
+        return self.arguments
+
+    # pylint: disable=no-self-use
```

Comment:
It uses self now with `_*_token`, you can remove the pylint exception.

Reply:
Removed.